### ISSUE TITLE
Generate nightly build for MacOS 10.13

### ISSUE THREAD

**Issue by gabriel-viviani** [neovim/neovim#13512]
- `nvim --version`: [Nightly 0.5.0](https://github.com/neovim/neovim/releases/tag/nightly)
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version: MacOS 10.13.06
- Terminal name/version: zsh

### Steps to reproduce using `nvim -u NORC`
 * Download nightly  for MacOs `.tar`
 * run `./nvim`

### Actual behaviour
Binary Incompatibility shown by:

```bash
./nvim
dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/gabrielviviani/Downloads/nvim-osx64/bin/./nvim
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/gabrielviviani/Downloads/nvim-osx64/bin/./nvim
  Expected in: /usr/lib/libSystem.B.dylib
```
![image](https://user-images.githubusercontent.com/19916771/101911158-c1f71880-3b9e-11eb-9324-4a22eff37601.png)

### Expected behaviour
Running nightly release.

IMHO what's needed is generating a compatible version for this OS version.
I'm naive and actually don't know/have time to do that build.

**Comment by vigoux** [comment]
This is related to #13312

cc @jamessan

**Comment by gabriel-viviani** [comment]
> This is related to #13312
> 
> cc @jamessan

Should I close this one or it'll be closed when PR is done?

**Comment by jamessan** [comment]
We already have nightly builds happening, and we're building with `CMAKE_OSX_DEPLOYMENT_TARGET` set to 10.11 so I would expect it to be compatible with newer releases.

I'll need someone with more macOS knowledge to help out with this.

**Comment by EduardoRFS** [comment]
This happens because it was built using a newer SDK definitely, macOS is only binary backwards compatible, not forwards compatible. So a binary built on Catalina will not run on High Sierra, but a binary built on High Sierra will run on Catalina(and big sur)

**Comment by gabriel-viviani** [comment]
Isn't [this](https://github.com/mhanberg/neovim/blob/81aa7809096568ed034bd1b89f03522b631885e0/.github/workflows/release.yml#L71) suppose to do what are you saying @EduardoRFS ?

**Comment by fsouza** [comment]
afaik GHA only support Catalina and Big Sur (10.15 or 11.0): https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

Using an older SDK on 10.15 should be possible, but I don't know enough about this either.

**Comment by EduardoRFS** [comment]
@gabriel-viviani it should be, but is that the build that you're using? Because as @fsouza mentioned GHA is only Catalina and Big Sur

**Comment by jamessan** [comment]
According to the [virtual environment](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#xcode) information, only xcode for 10.14 or newer is available by default.

We would need to install an older xcode version.  If someone provides instructions, I can plug it into the release build.  (Or provide a PR).

**Comment by jamessan** [comment]
@gabriel-viviani Could you try this nightly? https://github.com/jamessan/neovim/releases/download/nightly/nvim-macos.tar.gz

I changed the build to use the oldest available xcode.

**Comment by gabriel-viviani** [comment]
@jamessan unfortunately this was the output received: 
![image](https://user-images.githubusercontent.com/19916771/102082078-134a1680-3df0-11eb-9d8d-526ffb2d9863.png)
Basically same error
Does this means that even released nvim 0.5.0 won't be available for macOs older than 10.15?

**Comment by gabriel-viviani** [comment]
I did found [this site](www.xcodereleases.com) with Xcode releases, maybe latter on I could try to generate a compatible build.
Instructions from [source install] should be all I need to generate a nightly?

**Comment by jamessan** [comment]
Doesn't look like you can download XCode releases unless you're logged in to an Apple developer account. That's not going to work for CI.

It looks like our binary releases will be more limited in where they run, but I think Homebrew supports more versions.

**Comment by gabriel-viviani** [comment]
Well, just in case I generated one to myself.
It's available for my mac 10.13.16

[nvim-macos.tar.gz](https://github.com/neovim/neovim/files/5712897/nvim-macos.tar.gz)

@jamessan @vigoux @mhanberg

**Comment by jamessan** [comment]
Due to the macOS versions supported by GitHub Actions, this isn't something we'll be able to support.

If you need nvim on older macOS versions, you'll need to build yourself or use [Homebrew](https://brew.sh).
